benchmarks/checkpointing.ts

```diff
@@ -0,0 +1,60 @@
+import * as async from 'async'
+import * as crypto from 'crypto'
+var Trie = require('../dist/index.js').CheckpointTrie
+
+var iterations = 500
+var samples = 20
+var i
+
+function iterTest(numOfIter, cb) {
+  var vals = [] as any
+  var keys = [] as any
+
+  for (i = 0; i <= numOfIter; i++) {
+    vals.push(crypto.pseudoRandomBytes(32))
+    keys.push(crypto.pseudoRandomBytes(32))
+  }
+
+  var hrstart = process.hrtime()
+  var numOfOps = 0
+  var trie = new Trie()
+
+  for (i = 0; i < numOfIter; i++) {
+    trie.put(vals[i], keys[i], function () {
+      trie.checkpoint()
+      trie.get(Buffer.from('test'), function () {
+        numOfOps++
+        if (numOfOps === numOfIter) {
+          var hrend = process.hrtime(hrstart)
+          cb(hrend)
+        }
+      })
+    })
+  }
+}
+
+i = 0
+var avg = [0, 0]
+
+async.whilst(
+  function () {
+    i++
+    return i <= samples
+  },
+  function (done) {
+    iterTest(iterations, function (hrend) {
+      avg[0] += hrend[0]
+      avg[1] += hrend[1]
+
+      console.info('Execution time (hr): %ds %dms', hrend[0], hrend[1] / 1000000)
+      done()
+    })
+  },
+  function () {
+    console.info(
+      'Average Execution time (hr): %ds %dms',
+      avg[0] / samples,
+      avg[1] / 1000000 / samples,
+    )
+  },
+)
```

Comment:
I've no idea about these benchmarking files TBH, did you test this and are they working (now)?

If they do and if they do some useful things, can you give this an additional entry to the README?

Reply:
Yes they work for me, the files were already present (I only upgraded them to TS). I didn't look into how useful they are yet but it's on my task list to investigate after some promisify work :)